calculator.go

```diff
@@ -1,13 +1,94 @@
-// Package calculator provides a library for simple calculations in Go.
 package calculator
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+import (
+	"errors"
+	"math"
+	"strconv"
+	"strings"
+)
+
+// Add takes a variable number of float64 and returns their sum.
+func Add(nums ...float64) float64 {
+	sum := 0.0
+	for _, num := range nums {
+		sum += num
+	}
+
+	return sum
+}
+
+// Subtract takes a variable number of float64 and returns their difference.
+func Subtract(nums ...float64) float64 {
+	difference := nums[0]
+
+	for i := 1; i < len(nums); i++ {
+		difference -= nums[i]
+	}
+	return difference
+}
+
+// Multiply takes a variable number of float64 and returns their product.
+func Multiply(nums ...float64) float64 {
+	product := 1.0
+	for _, num := range nums {
+		product *= num
+	}
+	return product
+}
+
+// Divide takes a variable number of float64 and returns their quotient, or an error if it's not a valid division.
+func Divide(nums ...float64) (float64, error) {
+	dividend := nums[0]
+
+	for i := 1; i < len(nums); i++ {
+		if nums[i] == 0 {
+			return 0, errors.New("Cannot divide by zero")
+		}
+		dividend /= nums[i]
+	}
+
+	return dividend, nil
 }
 
-// Subtract takes two numbers and returns the result of subtracting the second
-// from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+// SqrRoot takes a number and returns its square root provided it has one, otherwise it returns an error.
+func SqrRoot(a float64) (float64, error) {
+	if a < 0 {
+		return 0, errors.New("cannot get square root of negative number")
+	}
+	return math.Sqrt(a), nil
+}
+
+// StringOperations takes in an expression as a string and parses it to evaluate it. Ex: "10 + 5" returns 15
+func StringOperations(operation string) (float64, error) {
+	// op := strings.ReplaceAll(operation, " ", "")
```

Comment:
One useful thing I've learned is never to comment out code. If I need the code, I mustn't comment it, and if I don't, there's no reason for it to be there. Commented-out code is confusing for readers ("Why is this commented out? Is it wrong? Is it optional? Is it a suggestion? A comment?")